### ISSUE TITLE
delight: stratify_delight_outcomes silently coerces non-boolean rare_mask

### ISSUE THREAD

**Issue by Svector-anu** [SlopDotCash/asi#1772]
`stratify_delight_outcomes` in `alberta_framework/core/delight.py` casts `rare_mask` to `jnp.bool_` unconditionally via `jnp.asarray(rare_mask, dtype=jnp.bool_)`, silently accepting any numeric array instead of requiring an actual boolean mask.

Reproduction (current tree):

```text
[0.  0.5] rare_success= 0 rare_failure= 1
[ 0. nan] rare_success= 0 rare_failure= 1
```

Fractional values, integers, and even `NaN` all coerce to a truthy/falsy stratum label instead of being rejected, corrupting the common/rare measurement strata with an invalid experiment mask instead of failing closed.

`stratify_delight_outcomes` is directly exported from both `alberta_framework/core/__init__.py` and the package root `alberta_framework/__init__.py`, and accepts caller-provided runtime `rare_mask` arrays — reachable through the public API with non-hardcoded inputs.

Fix incoming in a linked PR: convert without narrowing first, require `dtype == jnp.bool_`, then flatten — works under `jax.jit` since dtype is static during tracing.

**Comment by lalalune** [comment]
Resolved on main by #1874. The consolidated implementation preserves the intended contract (including maintained package tooling instead of rewriting historical outputs where applicable), and post-merge main CI run 32201150336 is fully green.
